dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -552,6 +563,25 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         return tokenDAO.findByUserId(user.getId());
     }
 
+    /**
+     * Construct a URL for Dockstore
+     * @param dockstoreScheme Dockstore scheme, e.g. http or https
+     * @param dockstoreHostName Dockstore host name, e.g. dockstore.org
+     * @param dockstorePort Dockstore port, e.g. 4200
+     */
+    private String createHostUrl(String dockstoreScheme, String dockstoreHostName, String dockstorePort) {
```

Comment:
The validity of this comment may change if you find a  function that does exactly what you want as suggested by Denis, but this is otherwise a good use case for a unit test.

* Does it work if port is empty
* If port is standard, e.g., 443 for https, does it leave it out of the generated url

Reply:
Also there is a method MetadataResource.createBaseURL() which does the same thing. And interestingly it uses getUIPort() instead of getPort().

So, sorry, for the scope creep, but I think you could move the existing method out of MetadataResource into a helper class, then invoke it here and from MetadataResource.

Reply:
MetadataResource.createBaseURL doesn't seem as robust as using the URL class. I suspect it may miss some edge cases...

Reply:
A quick look at the URL constructor shows a port 443 would be included for https, but I could have missed something...

Reply:
> MetadataResource.createBaseURL doesn't seem as robust as using the URL class. I suspect it may miss some edge cases...

That sounds right, but I think you make it robust and reusable rather than introducing a new method that does something similar. For example, what if somebody in the future needs this functionality -- which implementation do they use? It's not clear.

Reply:
I'd like to make updating and moving MetadataResource.createBaseURL to a helper class and using one method a separate ticket.
https://ucsc-cgl.atlassian.net/browse/DOCK-193

Reply:
Implemented URIHelper with a createBaseUrl method for use by MetadataResource and WorkflowResource